[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request

<!--(For example: This pull request adds checkstyle plugin).-->
close #9680

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 commented:
@EricGao888 hi bro,I have modified the test class, and the local test is available. Please check it。

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 commented:
@EricGao888  look this

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,76 @@
+
+import org.slf4j.Logger;
+
+/**
```
We do not need the comment here. People know it from the class name.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,76 @@
+ * hive log listener thread
+ */
+public class HiveSqlLogThread extends Thread {
+    /**
```
We don't need this comment either.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,76 @@
+     */
+    private final HiveStatement statement;
+    /**
+     * logger
```
We do not need to comment if people could easily understand it from field name.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,76 @@
+    @Override
+    public void run() {
+        if (statement == null) {
+            threadLogger.info("hive statement is null,end this log query!");
```
```suggestion
            threadLogger.info("hive statement is null, end this log query!");
```

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,76 @@
+            while (!statement.isClosed() && statement.hasMoreLogs()) {
+                for (String log : statement.getQueryLog(true, 500)) {
+                    //hive mapreduce log
+                    threadLogger.info(log);
```
What about changing the name to `hiveMapReduceLogger` and remove the comment?

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -294,6 +294,13 @@ private void sendAttachment(int groupId, String title, String content) {
     private String executeQuery(Connection connection, SqlBinds sqlBinds, String handlerType) throws Exception {
         try (PreparedStatement statement = prepareStatementAndBind(connection, sqlBinds)) {
             logger.info("{} statement execute query, for sql: {}", handlerType, sqlBinds.getSql());
+            //hive log listener
```
The same here, we could use a better name and remove the comment.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -303,6 +310,13 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
         int result = 0;
         for (SqlBinds sqlBind : statementsBinds) {
             try (PreparedStatement statement = prepareStatementAndBind(connection, sqlBind)) {
+                //hive log listener
```
The same as above.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * hive sql listener test
```
We do not need comment here.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+        HiveStatement statement = PowerMockito.mock(HiveStatement.class);
+        PowerMockito.when(statement.isClosed()).thenReturn(false);
+        PowerMockito.when(statement.hasMoreLogs()).thenReturn(true);
+        PowerMockito.when(statement.getQueryLog(true,500)).thenReturn(mockLog);
```
```suggestion
        PowerMockito.when(statement.getQueryLog(true, 500)).thenReturn(mockLog);
```

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+        PowerMockito.when(statement.getQueryLog(true,500)).thenReturn(mockLog);
+        try {
+            //print process log
+            HiveSqlLogThread queryThread = PowerMockito.spy(new HiveSqlLogThread(statement,LOGGER,taskExecutionContext));
```
```suggestion
            HiveSqlLogThread queryThread = PowerMockito.spy(new HiveSqlLogThread(statement, LOGGER, taskExecutionContext));
```

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+            HiveSqlLogThread queryThread = PowerMockito.spy(new HiveSqlLogThread(statement,LOGGER,taskExecutionContext));
+            queryThread.start();
+
+            Assert.assertEquals(taskExecutionContext.getTaskType(), "hive");
```
Verifying `task type` does not make too much sense here, you need to verify the method calls and your `mockLog`.

EricGao888 commented:
This is a good feature. Thanks @fengjian1129 for submitting the PR! I add some comments. Could you please also take a look when available? @SbloodyS Thanks!

SbloodyS commented:
To target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, just desc). It would not only connect issue to PR but also close issue automatically when PR is be closed. @fengjian1129

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 commented:
@EricGao888 hi bro, please check

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,71 @@
+                    hiveMapReduceLogger.info(log);
+
+                    List<String> appIds = LoggerUtils.getAppIds(log, hiveMapReduceLogger);
+                    //get sql task yarn's application_id
```
```suggestion
                    // get sql task yarn's application_id
```

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+import org.slf4j.LoggerFactory;
+
+
+@RunWith(PowerMockRunner.class)
```
Better avoid using `Powermock`

EricGao888 reviewed (approved):
Overall LGTM, except few nits.

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

EricGao888 commented:
@SbloodyS @zhongjiajie PTAL when available, thanks : )

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

devosend committed:
[Fix][UI] Fix bug where crontab special of month and year (#11661)

hiSandog committed:
[fix][python] Sql pattern add truncate. (#11666)

jieguangzhou committed:
[bug][python] Fix task switch error when default branch not defined last (#11606)

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

zhongjiajie committed:
[python] Auto location when create process definition (#11681)

* Add auto format when submit process definition from python api
* add UPDATING.md

close: #8254

qingwli committed:
[doc] Correct E2E Doc, fix WorkerGroupPage typo (#11629)

MichaelDeSteven committed:
[Improvement-11662][dao] fix foreach null items in mapper.xml

seedscoder committed:
[Fix-11633] The data type of the PostgreSQL database column is wrong. (#11633) (#11634)

SbloodyS committed:
[CI] Fix cluster test abnormal (#11688)

* fix cluster test abnormal

EricGao888 committed:
[Feature][Task Plugin] Support hive cli task plugin (#11651)

* Add hive cli task plugin

* Add docs for hive-cli task plugin

* Fix hive-cli task related front-end nits and checkstyle

* Fix exception handling logic

abzymeatsjtu committed:
[DOC] improve zk digest doc (#11703)

ruanwenjun committed:
[Bug] [Worker] Optimize the getAppId method to avoid worker OOM when kill task (#11701)

* Fix kill job may cause worker oom

ruanwenjun committed:
Refactor heart beat task, use json to serialize/deserialize (#11702)

* Refactor heart beat task, use json to serialize/deserialize

106umao committed:
optimize the process instance query, change the date time range (#11719)

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+            Thread.sleep(5000);
+            Assert.assertEquals(taskExecutionContext.getAppIds(), "application_1231_2323");
+
+        } catch (Exception e) {
```
You shouldn't catch `Exception e` here. If anything fails and throws exception in the `try` block, the exception will get caught and the UT will pass, which is not expected. If you want to verify a specific exception, you could use `@Test(expected = XXXException.class)` in `jUnit4`.

EricGao888 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/test/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThreadTest.java:
```diff
@@ -0,0 +1,68 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
```
There is no need to use `Powermock` in this UT. `Mockito` will be enough.

EricGao888 replied:
For clarification, what I meant was we should avoid using anything which comes from  `org.powermock`. You could use `org.mockito.xxx` instead.

fengjian1129 replied:
OK

caishunfeng committed:
reduce task dependencies (#11731)

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 commented:
@EricGao888 I fixed the code

Tianqi-Dotes committed:
[feature][API]feature: add task type list and enable user to add task type to fav (#11727)

* feature: add task type list and enable user to add task type to fav

* make DependentExecute.java variable protected

* add standalone server fileSet

kezhenxu94 committed:
Add Kubernetes configmap reload to all components (#11730)

kezhenxu94 committed:
Add Kubernetes configmap reload to all components (#11730)

fengjian1129 commented:
@EricGao888 I use the result chart to show this change
1.appids write t_ ds_ task_ Instance table and print MR processing log
<img width="956" alt="image" src="https://user-images.githubusercontent.com/35831367/188057628-4f5e57c2-93e8-445f-afaa-20092c440b8b.png">
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/35831367/188057666-0036e739-9d7c-4757-93bf-69994d0da502.png">
<img width="1364" alt="image" src="https://user-images.githubusercontent.com/35831367/188057930-4931d850-6ca3-4186-aab3-356f02be6101.png">

fengjian1129 commented:
@EricGao888 t_ds_task_instance table
<img width="884" alt="image" src="https://user-images.githubusercontent.com/35831367/188058211-38b66409-9584-4621-8cd8-013eb25cc308.png">

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

kezhenxu94 committed:
Update doc according to recent update of Spring / Swagger (#11755)

caishunfeng committed:
[Feature-11530] add state history for process instance (#11757)

* [Feature] add state history for process instance (#97)

* add state history for process instance

* upsertProcessInstance

* remove unuse method

* fix UT

Co-authored-by: caishunfeng <534328519@qq.com>

Amy0104 committed:
[Fix][UI] Add the task name entry when the workflow instance detail page goes to the task instance page. (#11761)

fuchanghai committed:
[fix-#11691] Add lz4-java in clickhouse datasource. (#11697)

fuchanghai committed:
[feature-#11689] change replaceAll to replace in method of tryExecuteSqlResolveColumnNames (#11696)

ChrisYuan committed:
[Fix-11217] [Doc] add postgresql config in doc: datasource-setting (#11326)

* add postgresql config for datasoruce-setting

* Update docs/docs/en/guide/howto/datasource-setting.md

Co-authored-by: Eric Gao <ericgao.apache@gmail.com>

hezean committed:
[Fix-11583][UI] Fix the API usage of gantt graph (#11642)

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Tianqi-Dotes committed:
[fix][API]fix task types sequence (#11775)

* fix task types sequence

* fix mapper table name t_ds_fav_task

* fix sequence

jieguangzhou committed:
[Feat][python] Add parameter environment to task (#11763)

jieguangzhou committed:
[DSIP-11][python] create workflows from YAML configuration (#11611)

fengjian1129 committed:
[Bug-11650][worker] #11650 fix SQL type task, stop task cause NPE (#11668)

zhongjiajie committed:
[python] Fix permission error when create related objects (#11783)

after we add permission to our codebase in dev branch, will got error
when create objects from pydolphinscheduler, because users with user_id
equal to `Integer.MAX_VALUE` is ordinary users which can not have access
to create unless we grant permission to the role manually. this patch
change dummyUser to admin user's user_id == 1 to fix it

hiSandog committed:
[k8s] Delete unnecessary configurations. (#11808)

wjf222 committed:
[Feature][Registry] Support etcd as registry  (#10981)

wjf222 committed:
[Feature][Registry] Support etcd as registry  (#10981)

fuchanghai committed:
Update dolphinscheduler_ddl.sql fix alert t_ds_user add time_zone is not correct (#11795)

SbloodyS committed:
[Doc] Remove re-upload file function in the 3.0.0's doc (#11804)

xdu-chenrj committed:
[DSIP-13][python] New mechanism file plugins to Python API (#11360)

hiSandog committed:
[bug][k8s] Change the config file keys in K8s to be the same as those in common.properties (#11812)

sekikn committed:
[Improvement] Replace commons-lang 2 function invocations with commons-lang3 (#11810)

* [Improvement] Replace commons-lang 2 function invocations with commons-lang3

ruanwenjun committed:
Remove `not use web service` comment in alert server (#11819)

SbloodyS committed:
fix hdfs defaultFs not working (#11823)

kezhenxu94 committed:
Bump up dependencies to fix cves (#11765)

kezhenxu94 committed:
Bump up dependencies to fix cves (#11765)

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml:
```diff
@@ -55,5 +55,10 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-common</artifactId>
+        </dependency>
+
```
Please don't add `dolphinscheduler-common` into task plugin module.

fengjian1129 replied:
I need to use the LoggerUtils.buildTaskId() method,Can I write the same method in the dolphinscheduler-task-api

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -155,6 +156,14 @@ public void handle() throws TaskException {
 
+            // get appIds
+            List<String> appIds = LogUtils.getAppIdsFromLogFile(taskExecutionContext.getLogPath());
+            logger.info("task log path:[{}],yarn appIds:[{}]", taskExecutionContext.getLogPath(), appIds);
```
```suggestion
            logger.info("Resolve appIds from task log: [{}]", appIds, taskExecutionContext.getLogPath());
```

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -291,6 +300,12 @@ private void sendAttachment(int groupId, String title, String content) {
             logger.info("{} statement execute query, for sql: {}", handlerType, sqlBinds.getSql());
+
+            if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
+                logger.info("execute sql type is [{}]",DbType.HIVE.getDescp());
```
```suggestion
                logger.info("The current sql task type is [{}], will start a demon thread to resolve the appIds", DbType.HIVE);
```

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -300,6 +315,13 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
         for (SqlBinds sqlBind : statementsBinds) {
             try (PreparedStatement statement = prepareStatementAndBind(connection, sqlBind)) {
+
+                if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
+                    logger.info("execute sql type is [{}]",DbType.HIVE.getDescp());
+
+                    HiveSqlLogThread queryThread = new HiveSqlLogThread(statement, logger,taskExecutionContext);
+                    queryThread.start();
+                }
```
Can we extract this into a private method?

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,82 @@
+    private final TaskExecutionContext taskExecutionContext;
+
+    public HiveSqlLogThread(PreparedStatement statement, Logger logger, TaskExecutionContext taskExecutionContext) throws SQLException {
+        this.statement = statement.unwrap(HivePreparedStatement.class);
```
super.setDeamon(true)

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,82 @@
+
+    @Override
+    public void run() {
+        String taskLogName = LoggerUtils.buildTaskId(taskExecutionContext.getFirstSubmitTime(),
+            taskExecutionContext.getProcessDefineCode(),
+            taskExecutionContext.getProcessDefineVersion(),
+            taskExecutionContext.getProcessInstanceId(),
+            taskExecutionContext.getTaskInstanceId());
+        taskExecutionContext.setTaskLogName(taskLogName);
+
+        // set the name of the current thread
+        Thread.currentThread().setName(taskLogName);
```
You can directly use the sql task's logger to print the log.

fengjian1129 replied:
<img width="782" alt="image" src="https://user-images.githubusercontent.com/35831367/189047832-a7ddbe6b-b667-4328-b35a-c9972983e283.png">
This is printed with the task log,If you do not set a unique taskLogName for this thread, the log content cannot be displayed on the web page

caishunfeng committed:
[Improvement] Add remote task model (#11767)

caishunfeng committed:
[Improvement] Add remote task model (#11767)

stalary committed:
[Bug](dependent) Dependent downstream trigger error when schedule cycle not day. (#11734)

* FIX: dependent

* FIX: version

* MOD: for review

fuchanghai committed:
[fix-#11815] fix ck column names contain special characters (#11818)

Co-authored-by: fuchanghai <changhai.fu@marketingforce.com>

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 commented:
@ruanwenjun hi bro, thank you for checking it for me. I have modified the code as required

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,75 @@
+                }
+            }
+        } catch (SQLException e) {
+            hiveMapReduceLogger.error("Failed to view hive log,exception:[{}]", e.getMessage());
```
```suggestion
            hiveMapReduceLogger.error("Failed to view hive log,exception:[{}]");
```

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/HiveSqlLogThread.java:
```diff
@@ -0,0 +1,75 @@
+                    //get sql task yarn's application_id
+                    if (!appIds.isEmpty()) {
+                        hiveMapReduceLogger.info("yarn application_id is {}",appIds);
+                        taskExecutionContext.setAppIds(String.join(",", appIds));
```
Why you don't stop this loop here? you want to rewrite the appId in next loop?

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -156,6 +158,14 @@ public void handle(TaskCallBack taskCallBack) throws TaskException {
+            Set<String> appIds = LogUtils.getAppIdsFromLogFile(taskExecutionContext.getLogPath());
+            logger.info("Resolve appIds: [{}] from task log: [{}]", appIds, taskExecutionContext.getLogPath());
+            if (!appIds.isEmpty()) {
+                taskExecutionContext.setAppIds(String.join(",", appIds));
+                setAppIds(String.join(",", appIds));
```
```suggestion
                String appIdString = String.join(",", appIds);
                taskExecutionContext.setAppIds(appIdString);
                setAppIds(appIdString);
```

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -306,13 +319,31 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
+     * @throws SQLException sqlException
+     */
+    private void sqlLogListener(PreparedStatement statement) throws SQLException {
+        if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
```
```suggestion
        if (DbType.HIVE.name() == sqlParameters.getType()) {
```

ruanwenjun commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -306,13 +319,31 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
+     */
+    private void sqlLogListener(PreparedStatement statement) throws SQLException {
+        if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
+            logger.info("The current sql task type is [{}], will start a demon thread to resolve the appIds", DbType.HIVE);
```
```suggestion
            logger.info("The current sql task type is HIVE, will start a daemon thread to resolve the appIds");
```

ruanwenjun commented:
@caishunfeng Please take a look.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -306,13 +319,31 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
+        if (DbType.HIVE == DbType.valueOf(sqlParameters.getType())) {
+            logger.info("The current sql task type is [{}], will start a demon thread to resolve the appIds", DbType.HIVE);
+
+            HiveSqlLogThread queryThread = new HiveSqlLogThread(statement, logger,taskExecutionContext);
```
I don't think it's a good way to create thread for log parsing, you can just get appIds after executing sql.

fengjian1129 replied:
The purpose here is to start the thread to print the progress log of MR, so that there is a progress log of HIVE SQL execution in the worker log file, and then we can obtain appids using the LogUtils.getAppids() method.

caishunfeng replied:
Is it possible to parse all the logs at once after execution?

fengjian1129 replied:
The screenshot of the print log is as follows:
<img width="821" alt="image" src="https://user-images.githubusercontent.com/35831367/189870359-16ba2b67-ece2-405a-97a7-4ffd7c8d37da.png">

In SqlTask.class ,Parse all logs and get appid as follows:
<img width="869" alt="image" src="https://user-images.githubusercontent.com/35831367/189877917-fe6f4e98-2bc3-406e-8e2b-009b5e2df76d.png">

fengjian1129 replied:
@caishunfeng You can review my latest code

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
Merge branch 'fengjian-20220819' of github.com:fengjian1129/dolphinscheduler into fengjian-20220819

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] #9680 add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

fengjian1129 committed:
delete variable 'String sql' is never read.

delete variable 'String sql'

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

fengjian1129 commented:
@ruanwenjun hi bro,review the code for me. thanks

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

fengjian1129 committed:
Merge branch 'dev' into fengjian-20220819

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
Merge remote-tracking branch 'upstream/dev' into dev

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

Commit:
[Improvement-#9680][task-plugin] add hive sql log listener, print MapReduce progress log and obtain application_id allows developers to better track tasks

kezhenxu94 commented on dolphinscheduler-api/pom.xml:
```diff
@@ -195,6 +195,11 @@
             <artifactId>hbase-noop-htrace</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-kubernetes-fabric8-config</artifactId>
+        </dependency>
+
```
Remove this, perhaps you do it wrong when resolving conflicts

kezhenxu94 commented on dolphinscheduler-bom/pom.xml:
```diff
@@ -258,6 +258,26 @@
                 <scope>import</scope>
             </dependency>
 
+            <!-- Etcd -->
+            <dependency>
+                <groupId>io.etcd</groupId>
+                <artifactId>jetcd-core</artifactId>
+                <version>${jetcd.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.etcd</groupId>
+                <artifactId>jetcd-test</artifactId>
+                <version>${jetcd.test.version}</version>
+                <scope>test</scope>
+            </dependency>
+            <dependency>
+                <groupId>io.grpc</groupId>
+                <artifactId>grpc-bom</artifactId>
+                <version>${io.grpc.version}</version>
+                <scope>import</scope>
+                <type>pom</type>
+            </dependency>
+
```
Remove these lines

kezhenxu94 commented on dolphinscheduler-dist/release-docs/licenses/LISCENSE-perfmark-api.txt:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
You didn't add any new dependencies, this file is unexpected, please remove or confirm where this dependency is brought

kezhenxu94 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java:
```diff
@@ -365,13 +379,31 @@ private String executeUpdate(Connection connection, List<SqlBinds> statementsBin
+     */
+    private void sqlLogListener(PreparedStatement statement) throws SQLException {
+        if (DbType.HIVE.name().equals(sqlParameters.getType())) {
+            logger.info("The current sql task type is HIVE, will start a demon thread to resolve the appIds");
```
```suggestion
            logger.info("The current sql task type is HIVE, will start a daemon thread to resolve the appIds");
```

fengjian1129 replied:
ok, I will modify it according to your suggestions. Can you help me merge the code after modification

fengjian1129 replied:
ok, I will modify it according to your suggestions. Can you help me merge the code after modification

MonsterChenzhuo commented:
Hi @fengjian1129  , I would like to ask if the current pr is still going on, if you have other things to do there and have no time to take care of, I hope to finish the follow-up content

MonsterChenzhuo commented:
Hi, I would like to ask if the current pr is still going on

fengjian1129 commented:
@SbloodyS @ruanwenjun @EricGao888 
hi bro.
Can you help me to modify this PR and merge it?  Because this PR is too far behind.Many friends still want to use this function

deamonJohn commented:
why is not merge this pr ？
